go/vt/topo/replication.go

```diff
@@ -157,3 +160,86 @@ func FixShardReplication(ctx context.Context, ts Server, logger logutil.Logger,
 	logger.Infof("All entries in replication graph are valid")
 	return nil
 }
+
+// UpdateShardReplicationFields updates the fields inside a topo.ShardReplication object.
+func (ts Server) UpdateShardReplicationFields(ctx context.Context, cell, keyspace, shard string, update func(*topodatapb.ShardReplication) error) error {
+	nodePath := path.Join(KeyspacesPath, keyspace, ShardsPath, shard, ShardReplicationFile)
```

Comment:
I don't think we need to do that any more, it was a leftover from old code.

Reply:
You mean now all the code everywhere uses only shard names in lower case and there is no possibility to pass shard names through a flag (or read it from topology)?

Reply:
You can pass shard names through a flag, but if you use the wrong case, it will just fail (with unknown shard error). Same as for keyspaces, if you use the wrong case, it will fail. That's just more consistent.

But in most cases, we go through ValidateShard (see above) which already does the ToLower. So all storage is already lower case.

Reply:
A bit more on this: all code path that alter the topo go through ValidateShard, or TabletComplete, which will both fix the shard name to lower case for keyranges. So we really guarantee everything is lower case for keyranges (like a0-b0). For non-keyranges, we will just keep the case, for all topo implementations alike.